pkg/cloud/aws/services/elb/loadbalancer.go

```diff
@@ -81,19 +81,18 @@ func (s *Service) GetAPIServerDNSName() (string, error) {
 func (s *Service) DeleteLoadbalancers() error {
 	s.scope.V(2).Info("Deleting load balancers")
 
-	// Get default api server spec
-	spec := s.getAPIServerClassicELBSpec()
+	// Get default api server name.
+	elbName := GenerateELBName(s.scope.Name(), v1alpha1.APIServerRoleTagValue)
 
-	// Describe or create.
-	apiELB, err := s.describeClassicELB(spec.Name)
-	if IsNotFound(err) {
-		return nil
-	}
-	if err != nil {
+	// Describe and delete if exists.
+	if _, err := s.describeClassicELB(elbName); err != nil {
+		if IsNotFound(err) {
+			return nil
+		}
 		return err
 	}
 
-	if err := s.deleteClassicELBAndWait(apiELB.Name); err != nil {
+	if err := s.deleteClassicELBAndWait(elbName); err != nil {
```

Comment:
if i'm understanding, describeClassicELB was returning (nil,nil) at some point?

Reply:
I don't think so, unless I missed some other PR, this just uses the elbName that we already have to delete it, in case there is one

Reply:
oh, i assumed the panic was that apiELB was nil and we were accessing .Name on it

Reply:
The original panic was on getting the spec (which we use to create the ELB) because the security group doesn't exist and it panics.

Reply:
can you add a test for this? it wasn't obvious that was the cause from this PR

Reply:
Added